Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -929,6 +929,28 @@ public function isArrayMultiLine($index)
     }
 
     /**
+     * Check if there is an anonymous class under given index.
+     *
+     * @param int $index
+     *
+     * @return bool
+     */
+    public function isAnonymousClass($index)
+    {
+        $token = $this[$index];
+
+        if (!$token->isClassy()) {
+            throw new \LogicException('No classy token at given index');
```

Comment:
I don't see any benefit in it.
The name is `$tokens->isLambda($index)`. "Is under given index a lambda function?"

> So here, to follow same principle

I would change the principle.

But even when not changing the principle, I would be pragmatic regarding interface/trait. 
Then I would apply the principle this way:
"OK, I have a classy element and I want to check is it a anonymous class. So calling it on non-classy token is providing an invalid argument, so an exception is rised."